python/prepare_release.py

```diff
@@ -0,0 +1,296 @@
+        # commit and push changes on master-update-version-$curtime branch
+        commit_all_changes_on_branch(project_name, project_version, upcoming_version_branch)
+        if not is_test:
+            push_branch(upcoming_version_branch)
+            resource_status = ResourceStatus.UPCOMING_VERSION_REMOTE
+        # create pull request
+        if not is_test:
+            create_pull_request_for_upcoming_version_branch(main_branch, repository, upcoming_version,
+                                                            upcoming_version_branch)
+            resource_status = ResourceStatus.PULL_REQUEST_CREATED
+    else:
+        print(f"### {project_version} is a patch release. Executing Patch release flow ### ")
+        # checkout release branch in release-X.Y format
+        checkout_branch(release_branch_name, is_test)
+
+        # change version info in configure.in file
+        update_version_in_configure_in(configure_in_path, project_version)
+        # execute "auto-conf " (Not with f flag!!!)
+        execute_autoconf_f()
+        # change version info in multi_extension.out
+        update_version_in_multi_extension_out(multi_extension_out_path, project_version)
+
+        if cherry_pick_enabled:
+            # list all pr's with backport labels
+            list_and_cherrypick_prs_with_backport_labels(earliest_pr_date, main_branch, release_branch_name, repository)
+
+        # commit all-changes
+        commit_all_changes_on_branch(project_name, project_version, release_branch_name)
+        # create and push release-$minor_version-push-$curTime branch
+        release_pr_branch = f"{release_branch_name}_{uuid.uuid4()}"
+        create_branch(release_pr_branch)
+        if not is_test:
+            push_branch(release_pr_branch)
+    return release_branch_name, upcoming_version_branch, f"{DISTRIBUTED_DIR_PATH}/{newly_created_sql_file}", \
+           resource_status
+
+
+def list_and_cherrypick_prs_with_backport_labels(earliest_pr_date, main_branch, release_branch_name, repository):
+    print(
+        f"### Getting all PR with backport label after {datetime.strftime(earliest_pr_date, '%Y.%m.%d %H:%M')}### ")
+    all_related_prs = get_prs(repository, earliest_pr_date, main_branch)
+    # get commits for selected prs with backport label
+    prs_with_backport = get_prs_by_label(all_related_prs, "backport")
+    print(f"### OK {len(prs_with_backport)} PRs with backport label found. PR list is as below### ")
+    for pr in prs_with_backport:
+        print(f"No:{pr.number} Title:{pr.title}\n")
+    # cherrypick all commits with backport label
+    print(f"Cherry-picking PRs on {release_branch_name}...")
+    cherry_pick_prs(prs_with_backport)
+    print(f"OK Cherry pick completed for all PRs on branch {release_branch_name}")
+
+
+def create_pull_request_for_upcoming_version_branch(main_branch, repository, upcoming_version,
+                                                    upcoming_version_branch):
```

Comment:
maybe we can re-order func args: repository, `repository, main_branch, upcoming_version_branch, upcoming_version`

Reply:
Fixed